src/core/plugin_loader.py

```diff
@@ -38,25 +41,28 @@ def load(directory="plugins", prefix="plugins", permissive=False):
             plugins.append(plugin)
             module_name = "{0}.{1}.plugin_settings".format(prefix, dir)
 
-            # Load settings module
-            plugin_settings = import_module(module_name)
-            validate_plugin_version(plugin_settings)
-
-            # Load hooks
-            hooks.append(load_hooks(plugin_settings))
-
-            # Check for workflow
-            workflow_check = check_plugin_workflow(plugin_settings)
-            if workflow_check:
-                settings.WORKFLOW_PLUGINS[workflow_check] = module_name
-                STAGE_CHOICES.append(
-                    (plugin_settings.STAGE, plugin_settings.PLUGIN_NAME)
-                )
-                ELEMENT_STAGES[
-                    plugin_settings.PLUGIN_NAME] = [plugin_settings.STAGE]
-
-            # Call event registry
-            register_for_events(plugin_settings)
+            try:
+                # Load settings module
+                plugin_settings = import_module(module_name)
+                validate_plugin_version(plugin_settings)
+
+                # Load hooks
+                hooks.append(load_hooks(plugin_settings))
+
+                # Check for workflow
+                workflow_check = check_plugin_workflow(plugin_settings)
+                if workflow_check:
+                    settings.WORKFLOW_PLUGINS[workflow_check] = module_name
+                    STAGE_CHOICES.append(
+                        (plugin_settings.STAGE, plugin_settings.PLUGIN_NAME)
+                    )
+                    ELEMENT_STAGES[
+                        plugin_settings.PLUGIN_NAME] = [plugin_settings.STAGE]
+
+                # Call event registry
+                register_for_events(plugin_settings)
+            except ModuleNotFoundError as e:
+                logger.error('Could not lod module, error: {e}'.format(e=e))
```

Comment:
This seems to be from another branch

Reply:
I'm not sure if this is from another branch as I did do some work on handling plugin loading errors for HPEs.